civihr_employee_portal/src/View/CustomFieldsMappingFixer.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+namespace Drupal\civihr_employee_portal\View;
+
+/**
+ * To match unknown table names and field names when importing views
+ * with custom fields and custom groups from CiviCRM.
+ * Replaces unknown fields names and unknown tables like:
+ * - exported [name_of_the_field]_[ID1] with existing [name_of_the_field]_[ID2]
+ * - exported [name_of_the_table]_[ID3] with existing [name_of_the_table]_[ID4]
+ * ... where ID1, ID2, ID3 & ID4 are different numbers created by different
+ * mechanisms thus producing different IDs.
+ */
+class CustomFieldsMappingFixer {
+
+  /**
+   * To fix mapping on Views import of CiviCRM custom fields
+   *
+   * @param view $view
+   *     the view that is being imported / recreated
+   */
+  public static function fixMapping($view) {
+    foreach ($view->display as $display) {
+      foreach ($display->display_options as &$displayOption) {
+        if (!is_array($displayOption)) {
+          continue;
+        }
+
+        self::processDisplayOptions($displayOption);
+      }
+    }
+  }
+
+  /**
+   * Process all display options for an exported view only when it is related
+   * to a field and the field has a number at the end of its name, which is
+   * the pattern for custom fields from CiviCRM
+   *
+   * @var array
+   *     Display option a exported view structure
+   */
+  private static function processDisplayOptions(&$displayOption) {
+    foreach ($displayOption as &$option) {
+      // only continue if this option is about a field
+      if (!isset($option['field'])) {
+        continue;
+      }
+
+      $optionIdParts = explode('_', $option['field']);
+      $finalIdPart = array_pop($optionIdParts);
+      if (count($optionIdParts) === 0 || !intval($finalIdPart)) {
+        continue;
+      }
+
+      self::processOption($option);
+    }
+  }
+
+  /**
+   * To process a display option for a field in view
+   *
+   * @var array $option
+   *     structure that contains an option for a field in view
+   */
+  private static function processOption(&$option) {
+    // checking if this table is part of the civicrm views integration
+    $tableName = self::getCiviCRMtableName($option['table']);
+    $prefix = $tableName ? self::getPrefix($tableName) : false;
+    if ( $prefix ) {
+      $option['table'] = $tableName;
+      $columns = self::getColumnsFromTable($prefix . $tableName);
+      // if the table where the current field belongs exists does not have it
+      // then we can search for similar fields in the same table and make replacements
+      if (!isset($columns[$option['field']])) {
+        $fieldNameBase = self::removeLastNumberOfString($option['field']);
+        $pos = array_search($fieldNameBase, $columns);
+        if ($pos !== FALSE) {
+          $option['field'] = array_search($columns[$pos], $columns);
+        }
+
+      }
+
```

Comment:
These empty lines after each closing `}` are a bit weird and are not improving the readability. What I meant with adding empty lines between logical blocks of code is something more or less like this:

```php
// checking if this table is part of the civicrm views integration
$tableName = self::getCiviCRMtableName($option['table']);
$prefix = $tableName ? self::getPrefix($tableName) : false;

if ( $prefix ) {
  $option['table'] = $tableName;
  
  $columns = self::getColumnsFromTable($prefix . $tableName);
  
  // if the table where the current field belongs exists does not have it
  // then we can search for similar fields in the same table and make replacements
  if (!isset($columns[$option['field']])) {
    $fieldNameBase = self::removeLastNumberOfString($option['field']);
    
    $pos = array_search($fieldNameBase, $columns);
    if ($pos !== FALSE) {
      $option['field'] = array_search($columns[$pos], $columns);
    }
  }
}
```